tests/tests_unit/test_api/test_files.py

```diff
@@ -429,6 +429,10 @@ def test_upload_with_external_id(self, cognite_client, mock_file_upload_response
         path = os.path.join(os.path.dirname(__file__), "files_for_test_upload", "file_for_test_upload_1.txt")
         cognite_client.files.upload(path, external_id="blabla", name="bla", data_set_id=42)
 
+    def test_upload_with_label(self, cognite_client, mock_file_upload_response):
+        path = os.path.join(os.path.dirname(__file__), "files_for_test_upload", "file_for_test_upload_1.txt")
+        cognite_client.files.upload(path, external_id="blabla", name="bla", data_set_id=42, labels=[Label("PUMP")])
```

Comment:
Let's add a note here about why/when this broke (versions affected). (Could also consider not passing anything but `name` & `label`)

(`ValueError: Could not parse label: {'external_id': 'foo-bar'}`)